File: 0003. Longest Substring Without Repeating Characters/Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> map(256, -1);
        int res = 0, st = -1;

        for (int i = 0; i < s.length(); ++i) {
            if (map[s[i]] > st)
                st = map[s[i]];
            map[s[i]] = i;
            res = max(res, i - st);
        }

        return res;
    }
};

File: 0003. Longest Substring Without Repeating Characters/Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        d, st, res = {}, -1, 0;
        for i in range(len(s)):
            if s[i] in d and d[s[i]] > st:
                st = d[s[i]]
            d[s[i]] = i
            res = max(res, i - st)
        return res;